### ISSUE TITLE
Lock State is always UNKNOWN

### ISSUE THREAD

**Issue by clearwave1** [rospogrigio/airbnk_mqtt#48]
Even though the Status sensor is being set correctly (most of the time it reads Locked), the state of the Lock (cover) is always UNKNOWN.

I looked at the code and don't see state being set anywhere and the async_update code does nothing.

I am assuming it is related but the button icons for open and close are the same.

**Comment by clearwave1** [comment]
@rospogrigio , @formatBCE - any thoughts?

**Comment by formatBCE** [comment]
I have no idea... Probably, the lock protocol changed during this year's. My lock is long gone, @rospogrigio abandoned this repo I guess...

**Comment by rospogrigio** [comment]
Actually I haven't abandoned this repo, but I'm still stuck with the problem I reported in #46 so my appliance is still not working and I've been having personal issues so I never had the time to dive deeply into it and fix it. BTW I don't know if you have any suggestion @formatBCE on how I can debug it... thank you.
If I get to have my device up and running again I can check if I have the same issue that you @clearwave1 are reporting.

**Comment by djk314** [comment]
Is it possible to force open or close without looking at the status ? When i use my real key it's appear Jammed, if i use original app to close or open it it work again, i don't need the status lock or close, just want to force it to close or open.

**Comment by rospogrigio** [comment]
> Is it possible to force open or close without looking at the status ? When i use my real key it's appear Jammed, if i use original app to close or open it it work again, i don't need the status lock or close, just want to force it to close or open.

Yes, you should comment out the `if self._device.curr_state == LOCK_STATE_OPERATING:` line (and outdent the following one) in cover.py, in async_open and async_close methods. Give it a try.

**Comment by formatBCE** [comment]
@rospogrigio sorry, looks like it's me who abandoned it. :(
I will try to help with ESP!

**Comment by djk314** [comment]
> > Is it possible to force open or close without looking at the status ? When i use my real key it's appear Jammed, if i use original app to close or open it it work again, i don't need the status lock or close, just want to force it to close or open.
> 
> Yes, you should comment out the `if self._device.curr_state == LOCK_STATE_OPERATING:` line (the following indented block) in cover.py, in async_open and async_close methods. Give it a try.

I'm on thx, i will tell if it worked.

**Comment by clearwave1** [comment]
While that is not my original issue, I have had my lock start showing JAMMED at least once a day as well.

You code above will not help this. Here is my code base on what @rospogrigio suggested:

"""
        if self._device.curr_state == LOCK_STATE_OPERATING:
            _LOGGER.warning("Operation already in progress: please wait")
            raise Exception("Operation already in progress: please wait")
            return
"""

The 3 quotes are how you comment out a multi-line block in Python. I have done this in both the open and close methods.

After I made the change I had another JAMMED state and the lock will now open anyway.

I hope this helps.

**Comment by clearwave1** [comment]
@rospogrigio - if you can let me know where the cover state should be updated, I can code and test on my side since I have a working lock and then make a PR for the changes.

**Comment by djk314** [comment]
This one work after a Jammed status :
 custom_device.py

def isjammed(self) -> bool | None:
if self.curr_state == LOCK_STATE_LOCKED:
return True
else:
return False
But statuts is locked if it's jammed, it work so i don't care status for now.
